api/src/main/java/de/unistuttgart/iaas/amyassist/amy/core/plugin/api/registry/Location.java

```diff
@@ -0,0 +1,65 @@
+/*
+ * This source file is part of the Amy open source project.
+ * For more information see github.com/AmyAssist
+ * 
+ * Copyright (c) 2018 the Amy project authors.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ * 
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at 
+ * 
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ * 
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * For more information see notice.md
+ */
+
+package de.unistuttgart.iaas.amyassist.amy.core.plugin.api.registry;
+
+/**
+ * Location object interface
+ * @author Benno Krauß
+ */
+public interface Location {
+
+    int getId();
+
+    String getZipCode();
+
+    void setZipCode(String zipCode);
+
+    String getCity();
+
+    void setCity(String city);
+
+    String getStreet();
+
+    void setStreet(String street);
+
+    int getHouseNumber();
+
+    void setHouseNumber(int houseNumber);
+
+    double getLongitude();
+
+    void setLongitude(double longitude);
+
+    double getLatitude();
+
+    void setLatitude(double latitude);
+
+    boolean isHome();
+
+    void setHome(boolean home);
+
+    boolean isWork();
```

Comment:
that is a special property that only on Location have true and should not set this way.

Reply:
only "one" location is what you mean, right?
This could definitely be solved in a more elegant way. Maybe an enum for every location object. Either "normal", "home" or "work". I'm open to suggestions. However, this would involve more complexity when saving the entity to the database. (Won't fix for now)

Reply:
how this is implemented is not the point, the information should not be on the public interface.
https://github.com/AmyAssist/Amy/pull/157/files#diff-015d47268b2d0df420cd897b3441a646R32 is a good way to get the information as a plugin